game-core/src/main/java/games/strategy/engine/framework/GameDataManager.java

```diff
@@ -175,11 +182,14 @@ static void saveGame(
           data.releaseReadLock();
         }
       }
-    });
 
-    // now write to file
-    try (OutputStream zippedOut = new GZIPOutputStream(sink)) {
-      zippedOut.write(bytes);
+      // now write to sink (ensure sink is closed per method contract)
+      try (InputStream is = new FileInputStream(tempFile);
+          OutputStream os = new BufferedOutputStream(sink)) {
+        IOUtils.copy(is, os);
```

Comment:
Instead of reading and writing to a file, wouldn't it be more efficient to just move the file into the other? 
Depending on the OS this would even make use of journaling file systems, so the old savegame won't be corrupted by a power loss during this process. 
Not sure how much java works like the `mv` command in this regard, but it looks like [with the correct flags, this could be a one-liner](https://docs.oracle.com/javase/10/docs/api/java/nio/file/Files.html#move(java.nio.file.Path,java.nio.file.Path,java.nio.file.CopyOption...))

Reply:
Yes, moving the temporary file to the final destination instead of copying would be optimal.  However, that's only true if the final destination _is_ a file.

I was probably a little loose with the term "file" in the PR description.  While it is probably the most _common_ case to save a game to a file, it's not the _only_ case.  For example, one may send a save game over the network.  To support the non-file cases, `GameDataManager#saveGame()` accepts an `OutputStream` (the `sink` parameter above)--not anything that could be converted to a `Path`.

Reply:
But wouldn't it make sending a game over the network incredibly slow then, if the file has to be written and read first?

Reply:
I'm neutral on this then.
Waiting for @ron-murhammer's or @DanVanAtta's opinion on this.